### PR TITLE
Docs front page CSS cleanup

### PR DESCRIPTION
**Pull request by alexdahl-okta**
## Description:
* More consistency with (and inheritance from) site hero sections
* Language links changed from table to flexbox
  - Lines now wrap naturally, which makes a softer distinction between front- and back-ends. When we add Vue, large viewports will again see front- and back-ends on separate lines.
* Section backgrounds now appear full-width
* Visual tweaks and increased padding for content links
* Bonus: padding fixes for quickstarts

docs-front-20180322

### Resolves:
* [UX-1066](https://oktainc.atlassian.net/browse/UX-1066)

### CODE REVIEW

**Comment by robertjd** [comment]
LGTM 👍 can we add the Vue icon as part of this PR?  I think it will be useful to do in this PR so that we can get it lined up as you desire per the mocks in the Jira issue.

**Comment by alexdahl-okta** [comment]
@robertjd Happy to add Vue! That'd make this merge dependent on #1867 – are we okay with that?
(Adding Vue will be a simple one-line change to `_source/_assets/js/docsIndex.js`.)

File: _source/_assets/css/okta/pages/_docsIndex.scss [hunk 1]
```diff
@@ -1,123 +1,80 @@
...
-        font-weight: 400;
-		line-height: 30px;
-        margin: 0, 0px, 0, auto;
-		a {
-			display: inline-table;
+  #docs-languages {
+    margin: 40px auto 0;
+    max-width: 860px;
+    .docs-page-tiles {
+      display: flex;
```
[the remark this answers is not in the document]
Comment (alexdahl-okta) [reply _source/_assets/css/okta/pages/_docsIndex.scss]: Yeah, I don't think we support non-flex browsers. But I've got an idea to provide a more graceful fallback, so may make a small change.

File: _source/_assets/js/docsIndex.js [hunk 2]
```diff
@@ -1,52 +1,36 @@
 (function($) {
   // Handle building the icons and code languages
-  var frontend = [
+  var codeLanguages = [
     { name: 'android', label: 'Android' },
     { name: 'angular', label: 'Angular' },
     { name: 'react',label: 'React' },
-    { name: 'javascript', label: 'JavaScript' },
     { name: 'ios',  label: 'iOS' },
```
[the remark this answers is not in the document]
Comment (alexdahl-okta) [reply _source/_assets/js/docsIndex.js]: Could, but I'd prefer to wait for other language page updates...

Changing to Swift is the plan, but for consistency Android should move to be a child of Java. (And while we'll still include Angular and React here, they should move to be children of JS.) So just keeping scope here small.

**Pull request merged**